src/module.c

```diff
@@ -848,6 +848,64 @@ int AlterIndexCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
   }
 }
 
+// FT.ALIASADD <NAME> <TARGET>
+static int AliasAddCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  ArgsCursor ac = {0};
+  ArgsCursor_InitRString(&ac, argv + 1, argc - 1);
+  if (argc != 3) {
+    return RedisModule_WrongArity(ctx);
+  }
+  IndexLoadOptions loadOpts = {
+      .name = {.rstring = argv[2]},
+      .flags = INDEXSPEC_LOAD_NOALIAS | INDEXSPEC_LOAD_KEYLESS | INDEXSPEC_LOAD_KEY_RSTRING};
+  IndexSpec *sptmp = IndexSpec_LoadEx(ctx, &loadOpts);
+  if (!sptmp) {
+    return RedisModule_ReplyWithError(ctx, "Unknown index name (or name is an alias itself)");
+  }
+  QueryError status = {0};
+  int rc = IndexAlias_Add(RedisModule_StringPtrLen(argv[1], NULL), sptmp, 0, &status);
+  if (rc != REDISMODULE_OK) {
+    return QueryError_ReplyAndClear(ctx, &status);
+  } else {
+    return RedisModule_ReplyWithSimpleString(ctx, "OK");
+  }
+}
+
+static int AliasDelCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  if (argc != 2) {
+    return RedisModule_WrongArity(ctx);
+  }
+  IndexLoadOptions lOpts = {.name = {.rstring = argv[1]},
+                            .flags = INDEXSPEC_LOAD_KEYLESS | INDEXSPEC_LOAD_KEY_RSTRING};
+  IndexSpec *sp = IndexSpec_LoadEx(ctx, &lOpts);
+  if (!sp) {
+    return RedisModule_ReplyWithError(ctx, "Alias does not exist");
+  }
+  QueryError status = {0};
+  if (IndexAlias_Del(RedisModule_StringPtrLen(argv[1], NULL), sp, 0, &status) != REDISMODULE_OK) {
+    return QueryError_ReplyAndClear(ctx, &status);
+  } else {
+    return RedisModule_ReplyWithSimpleString(ctx, "OK");
+  }
+}
+
+static int AliasUpdateCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  if (argc != 3) {
+    return RedisModule_WrongArity(ctx);
+  }
+  IndexLoadOptions lOpts = {.name = {.rstring = argv[1]},
+                            .flags = INDEXSPEC_LOAD_KEYLESS | INDEXSPEC_LOAD_KEY_RSTRING};
+  IndexSpec *spOrig = IndexSpec_LoadEx(ctx, &lOpts);
+  if (spOrig) {
+    QueryError status = {0};
+    if (IndexAlias_Del(RedisModule_StringPtrLen(argv[1], NULL), spOrig, 0, &status) !=
+        REDISMODULE_OK) {
+      return QueryError_ReplyAndClear(ctx, &status);
+    }
+  }
+  return AliasAddCommand(ctx, argv, argc);
```

Comment:
could it be that this will failed? if no then assert, if yes then its a problem cause we deleted the alias from the old index but did not set it to new one.. do you think we should role back?

Reply:
Good catch. Done.